edx-platform/pearson-kaust-theme/lms/templates/navigation/navbar-logo-header.html

```diff
@@ -8,6 +8,9 @@
 <%
 self.real_user = getattr(user, 'real_user', user)
 username = self.real_user.username
+
+if getattr(self.real_user, 'profile', None):
+  username = self.real_user.username if not self.real_user.profile.name else self.real_user.profile.name
```

Comment:
Why don't  you just use an inline if?. If self.real_user isn't defined, shouldn't self.real_user.profile.name be empty? @Squirrel18

Reply:
I don't understand your question, but self.real_user is always defined... 'profile' is what's not always defined. @Jacatove

Reply:
Sorry I confused the names, but now my question is: Isn't self.profile always defined when creating the User, at least with empty attributes (profile.name)?.

Reply:
What do you think about this, username is already defined with 'self.real_user.username'. @Squirrel18 
```suggestion
username = self.real_user.username
if getattr(self.real_user, 'profile', None):
  username = username if not self.real_user.profile.name else self.real_user.profile.name
```